packages/test/test-service-load/src/gcDataStores.ts

```diff
@@ -794,45 +794,80 @@ export class MultiCollabDataObject extends SingleCollabDataObject implements IGC
 		const partnerRunId2 = ((myRunId + halfClients + 1) % config.testConfig.numClients) + 1;
 		const partnerId1 = `client${partnerRunId1}`;
 		const partnerId2 = `client${partnerRunId2}`;
+
 		/**
-		 * Set up an event handler that listens for changes in the data object map meaning that a child data object
-		 * was referenced or unreferenced by a client.
+		 * Set up an event listener that will run / stop activity based on the activities of the partner.
+		 * If a partner referenced a data store or attachment blob, run activity on the corresponding local object.
+		 * If a partner unreferenced a data store or attachment blob, stop activity on the corresponding local object.
 		 */
-		const partnerActivityRunner = async (changed: IValueChanged, local: boolean) => {
+		const partnerActivityRunner = async (
+			changed: IValueChanged,
+			local: boolean,
+			activityObjectMap: SharedMap,
+			partnerIds: string[],
+		) => {
 			if (local) {
 				return;
 			}
 
-			// Only collaborate with two other partner clients. If we collaborate with all clients, there would be too
-			// many ops and we might get throttled.
-			if (!changed.key.startsWith(partnerId1) && !changed.key.startsWith(partnerId2)) {
+			// Collaborate with the partners clients specified in partnerIds.
+			let collaborate: boolean = false;
+			for (const partnerId of partnerIds) {
+				if (changed.key.startsWith(partnerId)) {
+					collaborate = true;
+					break;
+				}
+			}
+			if (!collaborate) {
 				return;
 			}
```

Comment:
I think this does the same thing:
```suggestion
			// Collaborate with the partner clients specified in partnerIds.
			if (!partnerIds.some((partnerId) => changed.key.startsWith(partnerId))) return;
```

Reply:
Nice! TIL.